# Продажи с сайта амазон

Данные взяты с сайта kaggle.com: https://www.kaggle.com/datasets/karkavelrajaj/amazon-sales-dataset


## Датасет содержит в себе данные рейтингов и обзоров более тысячи продуктов Амазон в соответствии с информацией, указанной на официальном сайте Амазон.

Описание данных:

- `product_id` - айди продукта
- `product_name` - наименование продукта
- `category` - категория
- `discounted_price` - цена со скидкой
- `actual_price` - регулярная цена
- `discount_percentage` - процент скидки
- `rating` - рейтинг
- `rating_count` - количество проголосовавших людей
- `about_product` - описание продукта
- `user_id` - айди польщователя, написавшего обзор 
- `user_name` - имя пользователя, написавшего обзов 
- `review_id` - айди ревью
- `review_title` - короткая версия ревью
- `review_content` - полная версия ревью
- `img_link` - ссылка на картинку продукта
- `product_link` - ссылка на продукт на сайте амазон

In [1]:
import pandas as pd 
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
df = pd.read_csv('C:\\datasets\\amazon.csv')

In [3]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%    4.0       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  \
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...   
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...   
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...   
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...   
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...   

                                        review_title  \
0  Satisfied,Charging is really fast,Value for mo...   
1  A Good Braided Cable for Your Type C Device,Go...   
2  Good speed for earlier versions,Good Product,W...   
3  Good product,Good one,Nice,Really nice product...   
4  As good as original,Decent,Good one for second...   

                                      review_content  \
0  Looks durable Charging is fine tooNo complains...   
1  I ordered this cable to connect my phone to An...   
2  Not quite durable and sturdy,https://m.media-a...   
3  Good product,long wire,Charges good,Nice,I bou...   
4  Bought this instead of original apple, does th...   

                                            img_link  \
0  https://m.media-amazon.com/images/W/WEBP_40237...   
1  https://m.media-amazon.com/images/W/WEBP_40237...   
2  https://m.media-amazon.com/images/W/WEBP_40237...   
3  https://m.media-amazon.com/images/I/41V5FtEWPk...   
4  https://m.media-amazon.com/images/W/WEBP_40237...   

                                        product_link  
0  https://www.amazon.in/Wayona-Braided-WN3LG1-Sy...  
1  https://www.amazon.in/Ambrane-Unbreakable-Char...  
2  https://www.amazon.in/Sounce-iPhone-Charging-C...  
3  https://www.amazon.in/Deuce-300-Resistant-Tang...  
4  https://www.amazon.in/Portronics-Konnect-POR-1...

In [4]:
#сразу дропнем ненужные нам столбцы
df = df.drop(columns=['review_title', 'review_content', 'img_link', 'product_link'])

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1465 entries, 0 to 1464
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1465 non-null   object
 1   product_name         1465 non-null   object
 2   category             1465 non-null   object
 3   discounted_price     1465 non-null   object
 4   actual_price         1465 non-null   object
 5   discount_percentage  1465 non-null   object
 6   rating               1465 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1465 non-null   object
 9   user_id              1465 non-null   object
 10  user_name            1465 non-null   object
 11  review_id            1465 non-null   object
dtypes: object(12)
memory usage: 137.5+ KB


In [6]:
df.describe()

product_id                                       product_name  \
count         1465                                               1465   
unique        1351                                               1337   
top     B07JW9H4J1  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
freq             3                                                  5   

                                                 category discounted_price  \
count                                                1465             1465   
unique                                                211              550   
top     Computers&Accessories|Accessories&Peripherals|...             ₹199   
freq                                                  233               53   

       actual_price discount_percentage rating rating_count  \
count          1465                1465   1465         1463   
unique          449                  92     28         1143   
top            ₹999                 50%    4.1        9,378   
freq            120                  56    244            9   

                                            about_product  \
count                                                1465   
unique                                               1293   
top     [CHARGE & SYNC FUNCTION]- This cable comes wit...   
freq                                                    6   

                                                  user_id  \
count                                                1465   
unique                                               1194   
top     AHIKJUDTVJ4T6DV6IUGFYZ5LXMPA,AE55KTFVNXYFD5FPY...   
freq                                                   10   

                                                user_name  \
count                                                1465   
unique                                               1194   
top     $@|\|TO$|-|,Sethu madhav,Akash Thakur,Burger P...   
freq                                                   10   

                                                review_id  
count                                                1465  
unique                                               1194  
top     R3F4T5TRYPTMIG,R3DQIEC603E7AY,R1O4Z15FD40PV5,R...  
freq                                                   10

In [7]:
df.isna().sum()

product_id             0
product_name           0
category               0
discounted_price       0
actual_price           0
discount_percentage    0
rating                 0
rating_count           2
about_product          0
user_id                0
user_name              0
review_id              0
dtype: int64

In [8]:
df[df['rating_count'].isnull()]

product_id                                       product_name  \
282  B0B94JPY2N  Amazon Brand - Solimo 65W Fast Charging Braide...   
324  B0BQRJ3C47  REDTECH USB-C to Lightning Cable 3.3FT, [Apple...   

                                              category discounted_price  \
282  Computers&Accessories|Accessories&Peripherals|...             ₹199   
324  Computers&Accessories|Accessories&Peripherals|...             ₹249   

    actual_price discount_percentage rating rating_count  \
282         ₹999                 80%    3.0          NaN   
324         ₹999                 75%    5.0          NaN   

                                         about_product  \
282  USB C to C Cable: This cable has type C connec...   
324  💎[The Fastest Charge] - This iPhone USB C cabl...   

                          user_id    user_name      review_id  
282  AE7CFHY23VAJT2FI4NZKKP6GS2UQ       Pranav   RUB7U91HVZ30  
324  AGJC5O5H5BBXWUV7WRIEIOOR3TVQ  Abdul Gafur  RQXD5SAMMPC6L

In [9]:
df = df.drop(index=[282, 324])

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1463 entries, 0 to 1464
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1463 non-null   object
 1   product_name         1463 non-null   object
 2   category             1463 non-null   object
 3   discounted_price     1463 non-null   object
 4   actual_price         1463 non-null   object
 5   discount_percentage  1463 non-null   object
 6   rating               1463 non-null   object
 7   rating_count         1463 non-null   object
 8   about_product        1463 non-null   object
 9   user_id              1463 non-null   object
 10  user_name            1463 non-null   object
 11  review_id            1463 non-null   object
dtypes: object(12)
memory usage: 148.6+ KB


In [11]:
df.duplicated().sum()

72

In [12]:
df.drop_duplicates(inplace=True)

In [13]:
df.duplicated().sum()

0

In [14]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             ₹399   
1  Computers&Accessories|Accessories&Peripherals|...             ₹199   
2  Computers&Accessories|Accessories&Peripherals|...             ₹199   
3  Computers&Accessories|Accessories&Peripherals|...             ₹329   
4  Computers&Accessories|Accessories&Peripherals|...             ₹154   

  actual_price discount_percentage rating rating_count  \
0       ₹1,099                 64%    4.2       24,269   
1         ₹349                 43%    4.0       43,994   
2       ₹1,899                 90%    3.9        7,928   
3         ₹699                 53%    4.2       94,363   
4         ₹399                 61%    4.2       16,905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...  
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...  
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...  
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...  
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...

In [15]:
df.discounted_price = df.discounted_price.str.replace('₹', '') 
df.discounted_price = df.discounted_price.str.replace(',', '') 
df.actual_price = df.actual_price.str.replace('₹', '')
df.actual_price = df.actual_price.str.replace(',', '')
df.discount_percentage = df.discount_percentage.str.replace('%', '')
df.rating_count = df.rating_count.str.replace(',' , '')
df.rating = df.rating.str.replace(',' , '')

In [16]:
def percents(value):
    value /= 100
    return value

In [17]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 1391 entries, 0 to 1464
Data columns (total 12 columns):
 #   Column               Non-Null Count  Dtype 
---  ------               --------------  ----- 
 0   product_id           1391 non-null   object
 1   product_name         1391 non-null   object
 2   category             1391 non-null   object
 3   discounted_price     1391 non-null   object
 4   actual_price         1391 non-null   object
 5   discount_percentage  1391 non-null   object
 6   rating               1391 non-null   object
 7   rating_count         1391 non-null   object
 8   about_product        1391 non-null   object
 9   user_id              1391 non-null   object
 10  user_name            1391 non-null   object
 11  review_id            1391 non-null   object
dtypes: object(12)
memory usage: 141.3+ KB


In [18]:
df.head(5)

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...              399   
1  Computers&Accessories|Accessories&Peripherals|...              199   
2  Computers&Accessories|Accessories&Peripherals|...              199   
3  Computers&Accessories|Accessories&Peripherals|...              329   
4  Computers&Accessories|Accessories&Peripherals|...              154   

  actual_price discount_percentage rating rating_count  \
0         1099                  64    4.2        24269   
1          349                  43    4.0        43994   
2         1899                  90    3.9         7928   
3          699                  53    4.2        94363   
4          399                  61    4.2        16905   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...  
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...  
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...  
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...  
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...

In [20]:
df.rating.value_counts()

4.1    232
4.2    216
4.3    215
3.9    120
4.4    117
4.0    116
3.8     85
4.5     69
4       52
3.7     41
3.6     34
3.5     26
4.6     16
3.3     15
3.4     10
4.7      6
3.1      4
4.8      3
3.2      2
2.8      2
3.0      2
5.0      2
2.3      1
|        1
2        1
3        1
2.6      1
2.9      1
Name: rating, dtype: int64

In [22]:
df.loc[df.rating == '|']

product_id                                       product_name  \
1279  B08L12N5H1  Eureka Forbes car Vac 100 Watts Powerful Sucti...   

                                               category  discounted_price  \
1279  Home&Kitchen|Kitchen&HomeAppliances|Vacuum,Cle...            2099.0   

      actual_price  discount_percentage rating  rating_count  \
1279        2499.0                 16.0      |         992.0   

                                          about_product  \
1279  No Installation is provided for this product|1...   

                                                user_id  \
1279  AGTDSNT2FKVYEPDPXAA673AIS44A,AER2XFSWNN4LAUCJ5...   

                                              user_name  \
1279  Divya,Dr Nefario,Deekshith,Preeti,Prasanth R,P...   

                                              review_id  
1279  R2KKTKM4M9RDVJ,R1O692MZOBTE79,R2WRSEWL56SOS4,R...

In [23]:
df.drop(index=1279, inplace=True)

In [24]:
df.discounted_price = df.discounted_price.astype('float64')
df.actual_price =  df.actual_price.astype('float64')
df.discount_percentage =df.discount_percentage.astype('float64')
df.rating_count = df.rating_count.astype('float64')
df.rating = df.rating.astype('float64')

In [25]:
df['discount_percentage'] = df['discount_percentage'].apply(percents)

In [26]:
df.head()

product_id                                       product_name  \
0  B07JW9H4J1  Wayona Nylon Braided USB to Lightning Fast Cha...   
1  B098NS6PVG  Ambrane Unbreakable 60W / 3A Fast Charging 1.5...   
2  B096MSW6CT  Sounce Fast Phone Charging Cable & Data Sync U...   
3  B08HDJ86NZ  boAt Deuce USB 300 2 in 1 Type-C & Micro USB S...   
4  B08CF3B7N1  Portronics Konnect L 1.2M Fast Charging 3A 8 P...   

                                            category  discounted_price  \
0  Computers&Accessories|Accessories&Peripherals|...             399.0   
1  Computers&Accessories|Accessories&Peripherals|...             199.0   
2  Computers&Accessories|Accessories&Peripherals|...             199.0   
3  Computers&Accessories|Accessories&Peripherals|...             329.0   
4  Computers&Accessories|Accessories&Peripherals|...             154.0   

   actual_price  discount_percentage  rating  rating_count  \
0        1099.0                 0.64     4.2       24269.0   
1         349.0                 0.43     4.0       43994.0   
2        1899.0                 0.90     3.9        7928.0   
3         699.0                 0.53     4.2       94363.0   
4         399.0                 0.61     4.2       16905.0   

                                       about_product  \
0  High Compatibility : Compatible With iPhone 12...   
1  Compatible with all Type C enabled devices, be...   
2  【 Fast Charger& Data Sync】-With built-in safet...   
3  The boAt Deuce USB 300 2 in 1 cable is compati...   
4  [CHARGE & SYNC FUNCTION]- This cable comes wit...   

                                             user_id  \
0  AG3D6O4STAQKAY2UVGEUV46KN35Q,AHMY5CWJMMK5BJRBB...   
1  AECPFYFQVRUWC3KGNLJIOREFP5LQ,AGYYVPDD7YG7FYNBX...   
2  AGU3BBQ2V2DDAMOAKGFAWDDQ6QHA,AESFLDV2PT363T2AQ...   
3  AEWAZDZZJLQUYVOVGBEUKSLXHQ5A,AG5HTSFRRE6NL3M5S...   
4  AE3Q6KSUK5P75D5HFYHCRAOLODSA,AFUGIFH5ZAFXRDSZH...   

                                           user_name  \
0  Manav,Adarsh gupta,Sundeep,S.Sayeed Ahmed,jasp...   
1  ArdKn,Nirbhay kumar,Sagar Viswanathan,Asp,Plac...   
2  Kunal,Himanshu,viswanath,sai niharka,saqib mal...   
3  Omkar dhale,JD,HEMALATHA,Ajwadh a.,amar singh ...   
4  rahuls6099,Swasat Borah,Ajay Wadke,Pranali,RVK...   

                                           review_id  
0  R3HXWT0LRP0NMF,R2AJM3LFTLZHFO,R6AQJGUP6P86,R1K...  
1  RGIQEG07R9HS2,R1SMWZQ86XIN8U,R2J3Y1WL29GWDE,RY...  
2  R3J3EQQ9TZI5ZJ,R3E7WBGK7ID0KV,RWU79XKQ6I1QF,R2...  
3  R3EEUZKKK9J36I,R3HJVYCLYOY554,REDECAZ7AMPQC,R1...  
4  R1BP4L2HH9TFUP,R16PVJEXKV6QZS,R2UPDB81N66T4P,R...

In [27]:
# посмотрим на уникальные категории
df.category.unique()

array(['Computers&Accessories|Accessories&Peripherals|Cables&Accessories|Cables|USBCables',
       'Computers&Accessories|NetworkingDevices|NetworkAdapters|WirelessUSBAdapters',
       'Electronics|HomeTheater,TV&Video|Accessories|Cables|HDMICables',
       'Electronics|HomeTheater,TV&Video|Televisions|SmartTelevisions',
       'Electronics|HomeTheater,TV&Video|Accessories|RemoteControls',
       'Electronics|HomeTheater,TV&Video|Televisions|StandardTelevisions',
       'Electronics|HomeTheater,TV&Video|Accessories|TVMounts,Stands&Turntables|TVWall&CeilingMounts',
       'Electronics|HomeTheater,TV&Video|Accessories|Cables|RCACables',
       'Electronics|HomeAudio|Accessories|SpeakerAccessories|Mounts',
       'Electronics|HomeTheater,TV&Video|Accessories|Cables|OpticalCables',
       'Electronics|HomeTheater,TV&Video|Projectors',
       'Electronics|HomeAudio|Accessories|Adapters',
       'Electronics|HomeTheater,TV&Video|SatelliteEquipment|SatelliteReceivers',
       'Computers&Acces

In [28]:
print('Количество уникальных категорий товаров: {}'.format(df.category.nunique()))

Количество уникальных категорий товаров: 211


И найдем пятерку самых дорогих товаров, самых больших скидок, товаров с самым высоким рейтингом и количеством голосов.  

In [29]:
df.sort_values(by='actual_price', ascending=False).head(5)

product_id                                       product_name  \
249   B09WN3SRC7  Sony Bravia 164 cm (65 inches) 4K Ultra HD Sma...   
255   B0BC8BQ432  VU 164 cm (65 inches) The GloLED Series 4K Sma...   
283   B0B3XXSB1K  LG 139 cm (55 inches) 4K Ultra HD Smart LED TV...   
1354  B09NS5TKPN  LG 1.5 Ton 5 Star AI DUAL Inverter Split AC (C...   
568   B08VB57558  Samsung Galaxy S20 FE 5G (Cloud Navy, 8GB RAM,...   

                                               category  discounted_price  \
249   Electronics|HomeTheater,TV&Video|Televisions|S...           77990.0   
255   Electronics|HomeTheater,TV&Video|Televisions|S...           54990.0   
283   Electronics|HomeTheater,TV&Video|Televisions|S...           47990.0   
1354  Home&Kitchen|Heating,Cooling&AirQuality|AirCon...           42990.0   
568   Electronics|Mobiles&Accessories|Smartphones&Ba...           37990.0   

      actual_price  discount_percentage  rating  rating_count  \
249       139900.0                 0.44     4.7        5935.0   
255        85000.0                 0.35     4.3        3587.0   
283        79990.0                 0.40     4.3        1376.0   
1354       75990.0                 0.43     4.3        3231.0   
568        74999.0                 0.49     4.2       27790.0   

                                          about_product  \
249   Resolution: 4K Ultra HD (3840 x 2160) | Refres...   
255   Resolution: 4K Ultra HD (3840x2160) | Refresh ...   
283   Resolution: 4K Ultra HD (3840x2160) | Refresh ...   
1354  Split AC with inverter compressor: variable sp...   
568   5G Ready powered by Qualcomm Snapdragon 865 Oc...   

                                                user_id  \
249   AF6Z2OYIXRPZJHVYN2MFKKYHPHFQ,AH5SAORYVUN5MGIBL...   
255   AHY6AK5LXBTGXDDXSU57ISMDW55Q,AGULFHMPCHCL32WCI...   
283   AFCWL3MX7BP2ZUDD37MEAENZDQ2A,AGGFXDLCFZMTLJJDR...   
1354  AGBYWFEGGX6QM6XB3ZPQADKKXAHA,AGU6LHFOEHAFE34AC...   
568   AEGEOVAES62OFGQTSPSDSQ5U7SHA,AGMBYTP5MS3JCQZ2N...   

                                              user_name  \
249   David P.,Dr. ATUL THAKUR,Shriniwas Patil,Tahir...   
255    UtdBuzz,Mukund Nagvenkar,S.R K.,Pruthvi Raj Goud   
283   Raj Mohammed,Samson,Haroon sayeed,P Imran Khan...   
1354  ABHISHEK KUMAR,Nageswar,Denzil,krushna,Brijesh...   
568                                 Mohammed Suhail,GVR   

                                              review_id  
249   R16HCZ0W1TRSMM,R12J7UKQ0FX3O9,R8729SR7LQFUU,R1...  
255   R2G4T57OLXDVPL,R3IQ8PWVTWENBY,RH6UHEBP622FT,R3...  
283   RC3ZLDRM8GA9T,RMDN4PSDM8SKK,R1YFAMDJ7P0SY3,R2W...  
1354  R2GZHWNGVMBJFG,R3L27H7N1WH5BG,R200QONLM29B4B,R...  
568                       R3R5DS04EXELTJ,R3JBXYOBYRX0A8

In [30]:
df.sort_values(by='discount_percentage', ascending=False).head(5)

product_id                                       product_name  \
695  B097C564GC  rts [2 Pack] Mini USB C Type C Adapter Plug, T...   
380  B0BF54LXW6  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
334  B0BF57RN3K  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
372  B0BF4YBLPX  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   
368  B0BF563HB4  Fire-Boltt Ninja Call Pro Plus 1.83" Smart Wat...   

                                              category  discounted_price  \
695  Computers&Accessories|Accessories&Peripherals|...             294.0   
380        Electronics|WearableTechnology|SmartWatches            1799.0   
334        Electronics|WearableTechnology|SmartWatches            1799.0   
372        Electronics|WearableTechnology|SmartWatches            1799.0   
368        Electronics|WearableTechnology|SmartWatches            1799.0   

     actual_price  discount_percentage  rating  rating_count  \
695        4999.0                 0.94     4.3        4426.0   
380       19999.0                 0.91     4.2       13937.0   
334       19999.0                 0.91     4.2       13937.0   
372       19999.0                 0.91     4.2       13937.0   
368       19999.0                 0.91     4.2       13937.0   

                                         about_product  \
695  【WIDER COMPATIBILITY】This USB C Female to USB ...   
380  Fire-Boltt is India' No 1 Wearable Watch Brand...   
334  Fire-Boltt is India' No 1 Wearable Watch Brand...   
372  Fire-Boltt is India' No 1 Wearable Watch Brand...   
368  Fire-Boltt is India' No 1 Wearable Watch Brand...   

                                               user_id  \
695  AGO5SRT3ESLNL5WTRQFD5BSIYPQA,AGGNODUMRADB3PQQT...   
380  AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...   
334  AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...   
372  AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...   
368  AEC6UDCEAUIBIFHGQDQ4KR67GC4A,AHRKSUOZXKKDERRY3...   

                                             user_name  \
695  Naga Sai Gadagottu,Neha Sharma,Amazon Customer...   
380  venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...   
334  venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...   
372  venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...   
368  venkatesh kg,Raghuram,YOGESH,Manan Parmar,Niku...   

                                             review_id  
695  R3CUNCZTU43JPP,RSO46BN8S4OSU,R2UD5D7T4DZRE5,R2...  
380  R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...  
334  R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...  
372  R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...  
368  R1PKIMKR1E8X8T,R23UV7ZBIEEZD3,RYRHNVDKS5RFY,RS...

In [31]:
df.sort_values(by='rating', ascending=False).head(5)

product_id                                       product_name  \
174   B0BP7XLX48  Syncwire LTG to USB Cable for Fast Charging Co...   
775   B09ZHCJDP1  Amazon Basics Wireless Mouse | 2.4 GHz Connect...   
1201  B0BQ3K23Y1  Oratech Coffee Frother electric, milk frother ...   
1299  B0B53DS4TF  Instant Pot Air Fryer, Vortex 2QT, Touch Contr...   
1145  B0BR4F878Q  Swiffer Instant Electric Water Heater Faucet T...   

                                               category  discounted_price  \
174   Computers&Accessories|Accessories&Peripherals|...             399.0   
775   Computers&Accessories|Accessories&Peripherals|...             499.0   
1201  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...             279.0   
1299  Home&Kitchen|Kitchen&HomeAppliances|SmallKitch...            4995.0   
1145  Home&Kitchen|Heating,Cooling&AirQuality|WaterH...            1439.0   

      actual_price  discount_percentage  rating  rating_count  \
174         1999.0                 0.80     5.0           5.0   
775         1000.0                 0.50     5.0          23.0   
1201         499.0                 0.44     4.8          28.0   
1299       20049.0                 0.75     4.8        3964.0   
1145        1999.0                 0.28     4.8       53803.0   

                                          about_product  \
174   This sturdy and durable cable made of tpe and ...   
775   Reliable wireless connection up to 10m|Advance...   
1201  -Make delicious milk foam creamer for your dri...   
1299  VOLTAGE (230 Volts): Instant Vortex Air Fryer ...   
1145  ✔Quick Electric Hot Water Tap Heating tube: hi...   

                                                user_id  \
174   AF7EOXYL5K36BDP6PXF6K2TL5TPA,AEN7NV2P5WNHM7EXC...   
775   AFLLEPVLIAH2DFSHAZ77KWFM72ZA,AHY2YZWK63CNZ626M...   
1201  AGRJZJKWKIE573KM5FWPOH4F7YCA,AEHV4VOLDQX5XYA42...   
1299  AGXJAYXZKJ6NCPSLX57MXJLQ3F6Q,AEQYSJWBP6DN2IV2L...   
1145  AEU7DVFEL43XZ6T4D572W2ZLBRKQ,AGWOH7CFDMUVW52NM...   

                                              user_name  \
174   Amazon Customer,kamal sahota,Pankaj,luvsmplcty...   
775   Rambeer kumar,Ramesh,Digambar Shelke,awadhesh ...   
1201  Amazon Customer,Neha Mehta,Nischal Agarwal,tha...   
1299  nikhil,Rituja Ray,Prasan Krishnan Karthikeyan,...   
1145                    Yogita g.,Amazon Customer,Mimsy   

                                              review_id  
174   R1L2JNO4Y3BHYF,R2346F22YLZ9IG,R3A4GAQTCPE5U7,R...  
775   R76XPXMKXLWKH,R23S77AWPH5FP5,RK7Q6W5FOPESC,R2X...  
1201  R3907SDNN9VR5Y,R1NNMXA39722T8,RXQNT49DKJ26S,R2...  
1299  R2FHIBV8JE4CTB,R315K0BCU0KVKO,RD129PA7KQQOR,R3...  
1145       R2WHW4PEF14WOD,R2DCCZWUGI0O0K,R1FA1HH6VL1RAL

In [32]:
df.sort_values(by='rating_count', ascending=False).head(5)

product_id                                       product_name  \
47   B014I8SSD0  Amazon Basics High-Speed HDMI Cable, 6 Feet - ...   
65   B014I8SX4Y  Amazon Basics High-Speed HDMI Cable, 6 Feet (2...   
12   B07KSMBL2H  AmazonBasics Flexible Premium HDMI Cable (Blac...   
684  B07KSMBL2H  AmazonBasics Flexible Premium HDMI Cable (Blac...   
400  B07GQD4K6L  boAt Bassheads 100 in Ear Wired Earphones with...   

                                              category  discounted_price  \
47   Electronics|HomeTheater,TV&Video|Accessories|C...             309.0   
65   Electronics|HomeTheater,TV&Video|Accessories|C...             309.0   
12   Electronics|HomeTheater,TV&Video|Accessories|C...             219.0   
684  Electronics|HomeTheater,TV&Video|Accessories|C...             219.0   
400  Electronics|Headphones,Earbuds&Accessories|Hea...             379.0   

     actual_price  discount_percentage  rating  rating_count  \
47          475.0                 0.35     4.4      426973.0   
65         1400.0                 0.78     4.4      426973.0   
12          700.0                 0.69     4.4      426973.0   
684         700.0                 0.69     4.4      426972.0   
400         999.0                 0.62     4.1      363713.0   

                                         about_product  \
47   Please select appropriate display resolution &...   
65   HDMI A Male to A Male Cable: Supports Ethernet...   
12   Flexible, lightweight HDMI cable for connectin...   
684  Flexible, lightweight HDMI cable for connectin...   
400  The perfect way to add some style and stand ou...   

                                               user_id  \
47   AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTI...   
65   AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTI...   
12   AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTI...   
684  AEYJ5I6JZZPOJB6MGWRQOHRQLPSQ,AFY5TVFOMVHGBPBTI...   
400  AF4MVO4JNFDEPWFKZO62OAJKRIWA,AHVPAXEWPATRASBKH...   

                                             user_name  \
47   Rishav Gossain,Shashank Mallamraju,Purnendu,gu...   
65   Rishav Gossain,Shashank Mallamraju,Purnendu,gu...   
12   Rishav Gossain,Shashank Mallamraju,Purnendu,gu...   
684  Rishav Gossain,Shashank Mallamraju,Purnendu,gu...   
400  tarun kumar,mahesh radheshyam tawari,Blackspad...   

                                             review_id  
47   R1FKOKZ3HHKJBZ,R2WNMZI1EXTA0H,RCA1M3W4RIXUR,R3...  
65   R1FKOKZ3HHKJBZ,R2WNMZI1EXTA0H,RCA1M3W4RIXUR,R3...  
12   R1FKOKZ3HHKJBZ,R2WNMZI1EXTA0H,RCA1M3W4RIXUR,R3...  
684  R1FKOKZ3HHKJBZ,R2WNMZI1EXTA0H,RCA1M3W4RIXUR,R3...  
400  R2DD2M5YARW7R2,R2M9ZYNGGV1ZLN,RNWNTRNLSJWSB,R3...

## Вывод

В результате этого небольшого исследования была проведена обработка исходных данных, а именно:
- удалены пропуски
- почищены дубликаты
- заменены некорректные типы данных
- заменены некорректные символы в столбцах с ценой 
- также мы выяснили, что товары в датасете представлены в 211 различных категориях и нашли пятерку самых дорогих товаров, самых больших скидок (первое место - аж 94%) , пятерку самых рейтинговых товаров и самое большое количество отзывов. 